### ISSUE TITLE
Print case class values from a List

### ISSUE THREAD

**Issue by staeff777** [bmc/scalasti#11]
Is it possible to access of getters of case classes while iterating through a list?

```scala
import org.clapper.scalasti.ST

case class Data(one:Int, two:Int)

object Test extends App{

  val templateString =
    """
      |<data.one>
      |<data.two>
      |-----------------------
      |<dataList:{data|<data.one>}>
      |-----------------------
      |<dataList:{data|<data>}>
    """.stripMargin

  val datalist =  List(Data(1,2),Data(1,2))

  val template = ST(templateString)
  template.add("data", datalist(0))
  template.add("dataList",datalist)

  println(template.render())
}
```
Output is
```
1
2
-----------------------
-----------------------
Data(1,2)Data(1,2)
```

Expected Output:
```
1
2
-----------------------
11
-----------------------
Data(1,2)Data(1,2)
```

**Comment by bmc** [comment]
This appears to be a bug. If I do the exact same thing in native StringTemplate (e.g., using a case class with `@BeanProperty`-defined fields), it works as expected. I'll look into it.

**Comment by bmc** [comment]
I'm looking into this one. I just finished updating StringTemplate to version 3.0.0, which completely changes the API to be immutable. I'll look into fixing this bug ASAP.

**Comment by bmc** [comment]
Okay, I added a specific unit test for this exact case, and it's fixed in version 3.0.1, which I'm just about to publish.

However, as of 3.0.0, Scalasti is now immutable, which means its API has changed some. To pick up this fix, you'll have to update to 3.0.1, which will mean changing your code wherever you use Scalasti. There are some guidelines on the main Scalasti web page (http://software.clapper.org/scalasti/). See the **Upgrading** section.

Fixing this problem required a fair amount of sleuthing and code rework. Backporting it to the 2.x version of Scalasti really isn't something I want to do.
